business/mesh.go

```diff
@@ -474,54 +472,68 @@ func (in *MeshService) resolveRemoteClustersFromSecrets() ([]Cluster, error) {
 	// Strictly speaking, this list may be incomplete: it's list of visible clusters for a control plane.
 	// But, for now, let's use it as the absolute "list of clusters in the mesh (excluding home cluster)".
 
-	// "Remote secrets" are created using the command `istioctl x create-remote-secret` which
-	// labels the secrets with istio/multiCluster=true. Let's use that label to fetch the secrets of interest.
-	secrets, err := in.k8s.GetSecrets(conf.IstioNamespace, "istio/multiCluster=true")
-	if err != nil {
-		if errors.IsForbidden(err) {
-			// A forbidden error means that we don't have privileges to list secrets in the Istio namespace.
-			// This may be because sysadmin may not want us to do that for security and probably
-			// because it is known that the environment is a single-cluster. So, return
-			// and empty list of clusters, avoid the warning error and use a trace log message.
-			log.Trace("Not enough privileges to list secrets with istio/multiCluster=true label.")
-			return []Cluster{}, nil
+	// Remote cluster secrets are mounted on the file system by the Kiali installer under the "/kiali-remote-cluster-secrets" directory.
+	// Each cluster has its own subdirectory, usually named the same as the cluster (though not necessarily). e.g. "/kiali-remote-cluster-secrets/<cluster name>"
+	// Cluster configs are found in a file whose name is the cluster name; e.g. "/kiali-remote-cluster-secrets/<cluster name>/<cluster name>"
+	// So walk the directory "/kiali-remote-cluster-secrets" to obtain all known remote cluster configs.
+
+	remoteClusterSecrets := make(map[string][]byte, 0)
+
+	secretDirs, err := os.ReadDir(remoteClusterSecretsDir)
+	if err == nil {
+		for _, sd := range secretDirs {
+			secretAbsDir := remoteClusterSecretsDir + "/" + sd.Name()
+			secretFiles, err := os.ReadDir(secretAbsDir)
```

Comment:
We're expecting a separate remote secret dir per remote secret? Won't there just be 1 remote secret per cluster?

Reply:
This has to do with the way the secrets are mounted on the file system by k8s. It's weird. I think its because there can be multiple configs in the secret itself. So the way the file system will be laid out is:

`/kiali-remote-cluster-secrets/<name of the cluster as defined in the pod mountVolume>/<name of the secret data as found in the Secret itself, which is the name of the cluster if created by istioctl>`

The way I coded this up, it should support 1 cluster per secret, or N clusters per secret (the idea being that perhaps I want a single Secret to contain all the information on all my clusters). I did not test it, but that's what this code should support today. This came for free just because of the way k8s lays out the file system by the way the volumes are declared - we just loop through all the files in the directory. Right now, you'll probably only see 1 secret data file per cluster subdirectory.

Reply:
Hmmm can the operator/helm chart combine all the secrets into a single dir with a projected volume? https://kubernetes.io/docs/concepts/storage/projected-volumes/. This is not very important though so if it's much work to change then don't bother.